Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -15,129 +15,113 @@ import TorusUtils
+        customAuth = try CustomAuth(config: config)
+        userInfo = nil
     }
 
     public func getCurrentFactorKey() throws -> String {
-        guard let factor = appState.factorKey else {
+        guard let factor = factorKey else {
             throw CoreKitError.factorKeyUnavailable
         }
         return factor
     }
 
     public func getDeviceMetadataShareIndex() throws -> String {
-        guard let shareIndex = appState.deviceMetadataShareIndex else {
+        guard let shareIndex = deviceMetadataShareIndex else {
             throw CoreKitError.notFound(msg: "share index not found")
         }
         return shareIndex
     }
 
-    // TODO: This should accept a param class instead
-    public func loginWithOAuth(loginProvider: LoginProviders, clientId: String, verifier: String, jwtParams: [String: String] = [:], redirectURL: String = "tdsdk://tdsdk/oauthCallback", browserRedirectURL: String = "https://scripts.toruswallet.io/redirect.html") async throws -> MpcKeyDetails {
-        if loginProvider == .jwt && jwtParams.isEmpty {
-            throw CoreKitError.notFound(msg: "jwt login should provide jwtParams")
-        }
+    public func loginWithOAuth(singleLoginParams: SingleLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerLogin(args: singleLoginParams)
 
-        let sub = SubVerifierDetails(loginType: .web,
-                                     loginProvider: loginProvider,
-                                     clientId: clientId,
-                                     verifier: verifier,
-                                     redirectURL: redirectURL,
-                                     browserRedirectURL: browserRedirectURL,
-                                     jwtParams: jwtParams
-        )
-        
-        let customAuth = CustomAuth(web3AuthClientId: option.Web3AuthClientId, aggregateVerifierType: .singleLogin, aggregateVerifier: verifier, subVerifierDetails: [sub], network: network.toTorusNetwork(), enableOneKey: true)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.singleVerifierResponse.userInfo.verifier, verifierId: loginResponse.singleVerifierResponse.userInfo.verifierId)
+        userInfo = loginResponse.singleVerifierResponse.userInfo
 
-        let userData = try await customAuth.triggerLogin()
-        return try await login(userData: userData)
+        return result
+    }
+
+    public func loginWithOAuth(aggregateLoginParams: AggregateLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerAggregateLogin(args: aggregateLoginParams)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifier, verifierId: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifierId)
```

Comment:
error should be thrown explicitly if userInfo.verifier is empty. Also it it verifier or aggregate verifier here?

Reply:
@ieow